mobile-device-management/src/main/java/org/odk/collect/mobiledevicemanagement/MDMConfigObserver.kt

```diff
@@ -0,0 +1,40 @@
+package org.odk.collect.mobiledevicemanagement
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.RestrictionsManager
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+
+/**
+ * Manages configuration changes from a mobile device management system.
+ *
+ * See android.content.APP_RESTRICTIONS in AndroidManifest for supported configuration keys.
+ */
+class MDMConfigObserver(
+    private val mdmConfigHandler: MDMConfigHandler,
+    private val restrictionsManager: RestrictionsManager,
+    private val context: Context
+) : DefaultLifecycleObserver {
+
+    private val restrictionsReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            mdmConfigHandler.applyConfig(restrictionsManager.applicationRestrictions)
```

Comment:
It's not quite clear from the docs (https://developer.android.com/reference/android/content/RestrictionsManager#getApplicationRestrictions()), but I believe `getApplicationRestrictions()` will read from disk so it should be executed in a background thread. That also goes for the logic in `applyConfig` and in `ProjectCreator`.

Reply:
Fixed.